main.go

```diff
@@ -67,7 +65,8 @@ func newApp() (context.Context, context.CancelFunc, *cli.App) {
 			},
 		},
 		Commands: []*cli.Command{
```

Comment:
nit: would it not be possible to add global flags like exoscale API keys and k8s connection creds to the global app flags instead of adding them individually by command?

Reply:
Yes, it's possible to add global flags. However, the way the CLI handles global flags (they have to be provided before the subcommand, e.g. `binary --global-flag subcommand`), I recommend to put each flag in a constructor/func like this: https://github.com/vshn/provider-exoscale/blob/master/operator_command.go#L34 and then use the func to provide the flag for each subcommand.
That way, you can do the following:
`binary subcommand --global-flag`
It allows still being able to control every flag for a subcommand (e.g. for some subcommands you might want to hide/disable certain "global" flags)

Reply:
yes, I believe, the way @ccremer mentioned is how it's done for the current commands. Unless I'm missing something else. Feel free to reopen the conversation again, if so.
Maybe you meant to add them as Flags on the `cli.App`? In that case, we'll need to read them from the context in the sub commands and I don't think we want that.

Reply:
as said, this is just a nit. 

To me, it does not make sense to have the same flags added individually to each subcommand, especially not essential ones like exoscale API keys (this is the exoscale metrics collector after all) and k8s url/token (which is required everywhere). 

The order of how to provide the flags to the command is, to me, neglectable since it's not a user facing command, we configure the cronjob once and be done with it. Also IIRC we provide the flags via env anyway so that's not an issue. 

Reading them from the context in the sub command: ok that's a bit inconvenient but neglectable IMO. 

but that's just my 2c on a relatively minor issue and shouldn't prevent the PR from getting merged.